Pcap++/src/PcapFilter.cpp

```diff
@@ -221,24 +216,54 @@ void IPFilter::convertToIPAddressWithLen(std::string& ipAddrmodified) const
 		addrAsInt &= mask;
 		ipAddrmodified = IPv4Address(addrAsInt).toString();
 	}
+	else if (ipAddr.getType() == IPAddress::IPv6AddressType)
+	{
+		std::array<uint8_t, 16> addrAsBytes{};
+		{
+			const uint8_t* bytesArrInternal = ipAddr.getIPv6().toBytes();
+			std::copy(bytesArrInternal, bytesArrInternal + 16, addrAsBytes.begin());
+		}
+
+		std::array<uint8_t, 16> mask{};
+		const int fullyMaskedBytes = m_Len / 8;
+		const int partialMaskBits = m_Len % 8;
+		for (size_t i = 0; i < static_cast<size_t>(fullyMaskedBytes) && i < mask.size(); i++)
+		{
+			mask[i] = static_cast<uint8_t>(-1);
+		}
+		if (static_cast<size_t>(fullyMaskedBytes) < mask.size())
+		{
+			mask[fullyMaskedBytes] = static_cast<uint8_t>(-1) << (8 - partialMaskBits);
+		}
+		for (size_t i = 0; i < addrAsBytes.size(); i++)
+		{
+			addrAsBytes[i] &= mask[i];
+		}
+
+		ipAddrmodified = IPv6Address(addrAsBytes.data()).toString();
+	}
+	else
```

Comment:
maybe remove the last `else{}` and move the `throw` outside, then we can save three lines.

Reply:
Are we tho? We need to add `return` statements at the end of each valid block, if we pull the `throw` outside the `if`/`else` chain.

Another thing to thing about, the `if`/`else` chain could potentially be refactored into a `switch`.

Reply:
you are correct, then it is fine.